scripts/zkai_attention_kv_multihead_quantized_softmax_receipt_gate.py

```
@@ -0,0 +1,782 @@
+        selected_scores = {row["selected_score"] for row in step_rows}
+        if len(selected_scores) != 1:
+            raise MultiheadQuantizedSoftmaxReceiptGateError(f"{profile.profile_id} selected score drift")
+        max_score = next(iter(selected_scores))
+        recomputed_denominator = 0
+        token_positions = {row["token_position"] for row in step_rows}
+        if len(token_positions) != 1:
+            raise MultiheadQuantizedSoftmaxReceiptGateError(f"{profile.profile_id} token-position drift")
+        query_token_position = next(iter(token_positions))
+        for row in step_rows:
+            if row.get("mask_allowed") not in (True, 1):
+                raise MultiheadQuantizedSoftmaxReceiptGateError(f"{profile.profile_id} mask row not allowed")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)